test/Microsoft.AspNet.Mvc.Core.Test/Filters/FormatFilterTest.cs

```diff
@@ -0,0 +1,253 @@
+        }
+
+        private static ResourceExecutingContext CreateResourceExecutingContext(
+            IFilter[] filters,
+            string format = null, 
+            FormatPlace? place = null)
+        {
+            if(format == null || place == null)
+            {
+                var context = new ResourceExecutingContext(
+                    CreateActionContext(),
+                    filters);
+                context.Result = new HttpStatusCodeResult(200);
+                return context;
+            }
+
+            var context1 = new ResourceExecutingContext(
+                CreateActionContext(format, place),
+                filters);
+            context1.Result = new HttpStatusCodeResult(200);
+            return context1;
+        }
+
+        private static ResultExecutingContext CreateResultExecutingContext(
+            string format = null,
+            FormatPlace? place = null)
+        {
+            if (format == null || place == null)
+            {
+                return new ResultExecutingContext(
+                    new ActionContext(new DefaultHttpContext(), new RouteData(), new ActionDescriptor()),
+                    new IFilter[] { },
+                    new ObjectResult("Some Value"));
+            }
+
+            return new ResultExecutingContext(
+                CreateActionContext(format, place),
+                new IFilter[] { },
+                new ObjectResult("Some Value"));
+        }
+
+        private static ActionContext CreateActionContext(string format = null, FormatPlace? place = null)
+        {
+            var httpContext = CreateMockHttpContext();
+
+            if (place == FormatPlace.RouteData || place == FormatPlace.RouteAndQueryData)
+            {
+                var data = new RouteData();
+                data.Values.Add("format", format);
+                httpContext.Setup(c => c.Request.Query.ContainsKey("format")).Returns(false);
+                return new ActionContext(httpContext.Object, data, new ActionDescriptor());
+            }
+
+            if (place == FormatPlace.QueryData || place == FormatPlace.RouteAndQueryData)
+            {
+                httpContext.Setup(c => c.Request.Query.ContainsKey("format")).Returns(true);
+                httpContext.Setup(c => c.Request.Query.Get("format")).Returns(format);
+                return new ActionContext(httpContext.Object, new RouteData(), new ActionDescriptor());
+            }
+            else if(place == null && format == null)
+            {
+                httpContext.Setup(c => c.Request.Query.ContainsKey("format")).Returns(false);
+                return new ActionContext(httpContext.Object, new RouteData(), new ActionDescriptor());
+            }
+
+            return null;
+        }
+
+        private static Mock<HttpContext> CreateMockHttpContext()
+        {
+            MvcOptions options = new MvcOptions();
+            MvcOptionsSetup.ConfigureMvc(options);
```

Comment:
Do you need to call setup from a unit test? Couldn't you register defaults and have the functional tests verify if the setup is working correctly?